sw/device/silicon_creator/lib/sigverify_unittest.cc

```diff
@@ -86,7 +87,48 @@ constexpr sigverify_rsa_buffer_t kEncMsg{
 constexpr std::array<uint8_t, 4> kSignedRegion{'t', 'e', 's', 't'};
 constexpr sigverify_rsa_buffer_t kSignature{};
 
-class SigVerifyTest : public mask_rom_test::MaskRomTest {
+/**
+ * Life cycle states used in parameterized tests.
+ */
+
+const std::array<lifecycle_state_t, 8> kLcStatesTest{
+    kLcStateTestUnlocked0, kLcStateTestUnlocked1, kLcStateTestUnlocked2,
+    kLcStateTestUnlocked3, kLcStateTestUnlocked4, kLcStateTestUnlocked5,
+    kLcStateTestUnlocked6, kLcStateTestUnlocked7,
+};
+
+const std::array<lifecycle_state_t, 4> kLcStatesNonTestOperational{
+    kLcStateDev,
+    kLcStateProd,
+    kLcStateProdEnd,
+    kLcStateRma,
+};
+
+const std::array<lifecycle_state_t, 12> kLcStatesNonOperational{
+    kLcStateRaw,         kLcStateTestLocked0,
+    kLcStateTestLocked1, kLcStateTestLocked2,
+    kLcStateTestLocked3, kLcStateTestLocked4,
+    kLcStateTestLocked5, kLcStateTestLocked6,
+    kLcStateScrap,       kLcStatePostTransition,
+    kLcStateEscalate,    kLcStateInvalid,
+};
+
+const std::unordered_set<lifecycle_state_t> kLcStatesAll = []() {
```

Comment:
Static initialization of containers is usually frowned upon due to initialization concerns. `constexpr` helps to mitigate some of the concerns, but ultimately this would require some refactoring. We should think about how to define this pattern as it will be pretty common in other unittests.

Reply:
Thanks for catching this, since this is actually forbidden by the styleguide. Fixed by moving this into a function and heap allocating to address concerns with initialization and destruction ([ref](https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use)), especially important if this pattern is used in other contexts. PTAL.